### PR TITLE
Narcolepsy Treatment

### PR DESCRIPTION
**Pull request by Scribbles0**
## About the PR 
This PR adds a new ReagentEffect to desoxyephedrine and ephedrine that resets the time between narcolepsy incidents. The preview video has an altered example in which the ReagentEffect resets the timer to 10. As soon as desoxyephedrine drops below 5, the entity will fall asleep 10 seconds later. In actual gameplay, the ReagentEffect will reset the timer to 600 (10 minutes).

This is my first time coding in C#, so please tell me if I need to change or fix anything.
**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://user-images.githubusercontent.com/91828755/211708018-5e330ad7-92a9-4b40-a3e8-e2985f1a3339.mp4

**Changelog**

:cl:
- add: Narcolepsy can now be treated.

### CODE REVIEW

File: Content.Server/Chemistry/ReagentEffects/ResetNarcolepsy.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+using Content.Server.Traits.Assorted;
+using Content.Shared.Chemistry.Reagent;
+using JetBrains.Annotations;
+
+namespace Content.Server.Chemistry.ReagentEffects
```
Comment (0x6273) [inline Content.Server/Chemistry/ReagentEffects/ResetNarcolepsy.cs:5]:
```suggestion
namespace Content.Server.Chemistry.ReagentEffects;
```
File-scoped namespaces are preferred for new .cs files.

**Comment by 0x6273** [comment]
You'll need to remove the curly brackets also and shift everything one indentation level to the left.

**Comment by Scribbles0** [comment]
> You'll need to remove the curly brackets also and shift everything one indentation level to the left.

where? i don't know which curly brackets you're asking me to delete as (i think) everything lines up

File: Content.Server/Chemistry/ReagentEffects/ResetNarcolepsy.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/// <summary>
+/// Reset narcolepsy timer
+/// </summary>
+[UsedImplicitly]
+public sealed class ResetNarcolepsy : ReagentEffect
+{
+    /// <summary>
+    /// The # of seconds the effect resets the narcolepsy timer to
+    /// </summary>
+    [DataField("TimerReset")]
```
Comment (metalgearsloth) [inline Content.Server/Chemistry/ReagentEffects/ResetNarcolepsy.cs:16]: This should be pascal-cased and using timespans.

**Pull request merged**